cbor.js

```diff
@@ -97,6 +229,34 @@ function encode(value) {
     }
   }
 
+  function writeUtf8String(value) {
```

Comment:
this function has been broken out of encodeItem and made to operate on the buffer directly